AsyncDisplayKit/ASNetworkImageNode.mm

```diff
@@ -333,6 +333,23 @@ - (void)fetchData
   }
 }
 
+#pragma mark - Layout and Sizing
+
+- (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
+{
+  ASDN::MutexLocker l(__instanceLock__);
+
+  // If the image node is currently in the loading process and no valid size is applied return CGSizeZero for the time
+  // being.
+  // TODO: After 2.0 is stable we should remove this behavior as a ASNetworkImageNode is a replaced element and the
+  // client code should set the size of an image or it's container it's embedded in
+  if (ASIsCGSizeValidForSize(constrainedSize) == NO && _URL != nil && self.image == nil) {
```

Comment:
`_URL == nil`: If `_URL` is nil in this case there is something wrong with the client code. The `_URL` should never be nil in this case, also if a `defaultImage` is set a `_URL` should be set.

`.defaultImage`: The `.defaultImage` size wasn't used in the size calculation in `1.x` so I think we should not introduce some new behavior I would like to avoid anyway.

In general: This overall solution should be there for a short period in the first place until the conversion pain points from 1.x to 2 are solved, because an `ASNetworkImageNode` always should have some kind of size set.